marketpulse/auth/backend.py

```diff
@@ -0,0 +1,42 @@
+from django.conf import settings
+from django.contrib.auth.models import User
+
+from marketpulse.base.mozillians import MozilliansClient
+
+from django_browserid.auth import BrowserIDBackend, default_username_algo
+
+
+def default_authentication_policy(user):
+    return True
+
+
+class MozilliansAuthBackend(BrowserIDBackend):
+    def __init__(self, *args, **kwargs):
+        super(MozilliansAuthBackend, self).__init__(*args, **kwargs)
+
+        api_url = settings.MOZILLIANS_API_URL
+        api_key = settings.MOZILLIANS_API_KEY
+        self.mozillians_client = MozilliansClient(api_url, api_key)
+
+    def is_valid_email(self, email):
+        try:
+            user = self.mozillians_client.lookup_user(email, detailed=False)
+        except:
+            user = None
+
+        authentication_policy = getattr(settings, 'MOZILLIANS_AUTHENTICATION_POLICY', None)
+        if authentication_policy:
+            return user and authentication_policy(user)
+        return user and default_authentication_policy(user)
+
+    def create_user(self, email):
+        try:
+            user = self.mozillians_client.lookup_user(email)
+        except:
+            user = None
+
+        if user:
+            email = user['email']['value']
+            username = default_username_algo(email)
+            user = User.objects.create_user(username=username, email=email)
+            return user
```

Comment:
No need for these, all you need to do is populate the UserProfile based on the email returned from browserid